### PR TITLE
Add #632

### LINKED ISSUE #632: Feature request: study list
OpenML API supports study listing, but I haven't implemented this in Python yet. Would be cool to add this.

https://test.openml.org/api/v1/study/list/status/all

Legal filters:
- status {in preparation, active, deactivated}
- uploader
- offset
- status

I believe the api docs are currently under review.

### PR DESCRIPTION
**Pull request by janvanrijn**
#### Reference Issue
Fixes #632

### CODE REVIEW

File: openml/study/functions.py [hunk 1]
```diff
@@ -278,3 +278,104 @@ def detach_from_study(study_id, entity_ids):
...
+    Return a list of all studies which are on OpenML.
+
+    Parameters
+    ----------
+    offset : int, optional
+        The number of studies to skip, starting from the first.
+    size : int, optional
+        The maximum number of studies to show.
+    main_entity_type : str, optional
+        Can be `task` or `run`. In case of `task`, only benchmark suites are
```
Comment (mfeurer) [inline openml/study/functions.py]: For rest you should format as:
```
``'task'`` or ``'run'``
```

File: openml/study/functions.py [hunk 2]
```diff
@@ -278,3 +278,104 @@ def detach_from_study(study_id, entity_ids):
...
+    assert study_dict['oml:study_list']['@xmlns:oml'] == \
+        'http://openml.org/openml', study_dict['oml:study_list']['@xmlns:oml']
+
+    studies = dict()
+    for study_ in study_dict['oml:study_list']['oml:study']:
+        expected_fields = {
+            'oml:id': 'id',
+            'oml:alias': 'alias',
+            'oml:main_entity_type': 'main_entity_type',
+            'oml:benchmark_suite': 'benchmark_suite',
```
Comment (mfeurer) [inline openml/study/functions.py]: Shouldn't this also be casted to an integer?

File: tests/test_study/test_study_functions.py [hunk 3]
```diff
@@ -18,13 +18,13 @@ def test_get_study(self):
         self.assertEqual(len(study.setups), 30)
 
     def test_get_tasks(self):
-        study_id = 14
+        study_id = 1
```
Comment (mfeurer) [inline tests/test_study/test_study_functions.py]: Could you briefly elaborate the issue with study 14?

Comment (janvanrijn) [reply tests/test_study/test_study_functions.py]: I removed it from the test server, and replaced it by study 1

**Pull request merged**